.github/workflows/release.yml

```diff
@@ -22,14 +22,121 @@ jobs:
+          echo "IMAGE_ORG=${IMAGE_ORG}" >> $GITHUB_ENV
+
+      - name: Set TAG from github ref name
         run: |
-          nix-shell --pure --run "./scripts/helm/update-version-upon-release.sh" ./shell.nix
+          TAG="${{ github.ref_name }}"
+          echo "TAG=${TAG}" >> $GITHUB_ENV
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Login to GHCR
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Mirror images from dev to Docker Hub
+        run: |
+          nix-shell --pure --run "./scripts/staging/mirror-images.sh \
+            --source-namespace ghcr.io/${{ env.IMAGE_ORG }}/dev \
+            --target-namespace docker.io/${{ secrets.DOCKERHUB_USERNAME }} \
+            --tag ${TAG}" ./shell.nix
+
+  release-kubectl-binaries:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          submodules: recursive
+
+      - uses: cachix/install-nix-action@v31.3.0
+      - name: Pre-populate nix-shell
+        run: |
+          export NIX_PATH=nixpkgs=$(jq '.nixpkgs.url' nix/sources.json -r)
+          echo "NIX_PATH=$NIX_PATH" >> $GITHUB_ENV
+          nix-shell --pure --run "echo" ./shell.nix
+
+      - name: Set Image Org
+        run: |
+          [ -z "${{ secrets.IMAGE_ORG }}" ] && IMAGE_ORG=openebs || IMAGE_ORG=${{ secrets.IMAGE_ORG }}
+          echo "IMAGE_ORG=${IMAGE_ORG}" >> $GITHUB_ENV
+
+      - name: Set TAG from github ref name
+        run: |
+          TAG="${{ github.ref_name }}"
+          echo "TAG=${TAG}" >> $GITHUB_ENV
+
+      - name: Download and Upload kubectl artifacts
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          nix-shell --pure --run "./scripts/staging/pull-kubectl-from-ghcr.sh \
+            --tag ${TAG} \
+            --namespace ghcr.io/${{ env.IMAGE_ORG }}/dev/kubectl-openebs \
+            --username ${{ github.actor }} \
+            --password ${{ secrets.GITHUB_TOKEN }}" ./shell.nix
+
+          ./mayastor/scripts/github/release-artifacts.sh \
+            --tag ${TAG} \
+            --repo-org ${{ github.repository_owner }} \
+            --upload ${{ github.event.repository.name }} \
+            upload
+
+  release-helm-chart:
+    runs-on: ubuntu-latest
+    needs: [release-images]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          submodules: recursive
+
+      - name: Set Image Org
+        run: |
+          [ -z "${{ secrets.IMAGE_ORG }}" ] && IMAGE_ORG=openebs || IMAGE_ORG=${{ secrets.IMAGE_ORG }}
+          echo "IMAGE_ORG=${IMAGE_ORG}" >> $GITHUB_ENV
```

Comment:
Can't we just use `github.repository_owner` ?

Reply:
Yes we can as I mentioned above.